packages/tooling/fast-tooling-react/src/form/custom-controls/css/color-picker-component/color-picker.ts

```diff
@@ -0,0 +1,400 @@
+     * @internal
+     */
+    public handleChange(): void 
+    {
+        this.$emit("change");
+    }
+
+    /**
+     * Determines if a number value is within the valid range for an R, G, or B color channel.
+     * @param val - Number to be evaluated.
+     */
+    private isValideRGB(val: number): boolean
+    {
+        return val >= 0 && val <= 255;
+    }
+
+    /**
+     * Determines if a number value is within the valid range for the alpha channel.
+     * @param val - Number to be evaluated.
+     */
+    private isValideAlpha(val: number): boolean
+    {
+        return val >= 0 && val <= 100;
+    }
+
+    /**
+     * Determines if a number value is within the valid range for the saturation or value color channels.
+     * @param val - Number to be evaluated.
+     */
+    private isValideSatVal(val: number): boolean
+    {
+        return val >= 0 && val <= 100;
+    }
+
+    /**
+     * Determines if a number value is within the valid range for the hue color channel.
+     * @param val - Number to be evaluated.
+     */
+    private isValideHue(val: number): boolean
+    {
+        return val >= 0 && val <= 359;
+    }
+
+    /**
+     * Checks if input is a valid CSS color.
+     * After placing an invalid css color value into a color style property the value will be an empty string when read back.
+     * @internal
+     */
+    private isValideCSSColor(testValue:string): boolean 
+    {
+        /* Set the background color of the proxy element since it is not visible in the UI. */
+        this.proxy.style.backgroundColor = "";
+        this.proxy.style.backgroundColor = testValue;
+        /* Read the value back out. If it was not a valid color value then it will be an empty string when read back out. */
+        return this.proxy.style.backgroundColor!="" ? true : false;
+    }
+
+    /**
+     * Update the current color values to a new HSV color.
+     * @param hue The new Hue value.
+     * @param sat The new saturation value.
+     * @param val The new Value value.
+     */
+    private updateHSV(hue:number, sat:number, val:number)
+    {
+        this.currentHSVColor = new ColorHSV(hue, sat, val);
+        this.currentRGBColor = hsvToRGB(this.currentHSVColor,this.currentRGBColor.a);
+        this.updateUIValues(true);
+    }
+
+    /**
+     * Update the current color values based on the mouse position over one of the three UI elements (hue, saturation/value, or alpha).
+     * @param pageX The pageX position of the mouse.
+     * @param pageY The pageY position of the mouse.
+     */
+    private updateFromMouseEvent(pageX:number, pageY:number)
+    {
+        let pos:DOMRect = this.currentMouseTarget.getBoundingClientRect();
+        var x = pageX - pos.left;
+		var y = pageY - pos.top;
+		var width = pos.width;
+        var height = pos.height;
+
+		if (x > width) x = width;
+		if (y > height) y = height;
+		if (x < 0) x = 0;
+		if (y < 0) y = 0;
+
+        if(this.currentMouseParam == 'h')
```

Comment:
nit: I know it's not necessary in these cases, but we typically use the `===` instead of `==` everywhere possible. I would suggest changing them.

```suggestion
        if(this.currentMouseParam === 'h')
```

Reply:
Done